edb/pgsql/compiler/relctx.py

```diff
@@ -959,11 +959,14 @@ def range_for_pointer(
 
 def rvar_for_rel(
         rel: Union[pgast.BaseRelation, pgast.CommonTableExpr], *,
-        lateral: bool=False, colnames: List[str]=[],
+        lateral: bool=False, colnames: Optional[List[str]]=None,
```

Comment:
I realize this goes contrary to some advice I recently gave. To reiterate that advice first: avoid Optionals if possible since None is viral. For instance, you should set arguments to related empty containers and be done with it.

However, this example connects two hairy edge cases of Python:
- setting mutable data structures as argument defaults; and
- the fact that a single string is a valid `Sequence[str]` (because strings are iterable).

We don't want to have a list as a default value because if a single call mutates it, the other calls will receive the mutated version.

On the other hand, we can't set a sensible immutable default instead because changing `List[str]` to `Sequence[str]` would undermine our ability to type check one of the most common errors: putting a single element in place of a list of elements.

So in the end we made the list optional :(

Reply:
Is there a linter for this? I.e. "don't use Sequence[str] as annotation"?

Reply:
Well, it's a valid annotation. It's just much less useful than people think. We might write our own linter for this if you think it's important.